pywbem_mock/_subscriptionproviders.py

```diff
@@ -191,20 +205,29 @@ def fix_key_properties(self, new_instance):
         """
         Fix the key properties that are common to many classes (SystemName,
         CreationClassName, SystemCreateionClassName) since these may be
-        either not provided with the new instance or incorrect for this
-        enviornment
+        either not provided,None, or incorrect value for this environment with
+        the new instance, they may be overridden. This works only for
+        the filter and destination classes that contain the same set of
+        key properties.
         """
         ccn_pname = 'CreationClassName'
         sys_pname = 'SystemName'
         sccn_pname = 'SystemCreationClassName'
 
         # Validate  or fix other key values in the new instance
         def _fix_key_prop(pname, test_value, replacement=None):
-            if pname not in new_instance or \
-                    new_instance[pname].lower != test_value.lower:
+            """
+            Replace property pname with either replacement if replacement not
+            None or test_value if property does not exist, is None, or not same
+            value as test_value
+            """
+            if pname not in new_instance or new_instance[pname] is None:
+                new_instance[pname] = replacement or test_value
+            elif new_instance[pname].lower != test_value.lower:
                 new_instance[pname] = replacement or test_value
```

Comment:
These two bodies could be combined into a single if.

Reply:
Done